src/ray/core_worker/lib/java/io_ray_runtime_task_NativeTaskSubmitter.cc

```diff
@@ -161,10 +161,13 @@ inline ActorCreationOptions ToActorCreationOptions(JNIEnv *env,
     }
     auto java_actor_lifetime = (jobject)env->GetObjectField(
         actorCreationOptions, java_actor_creation_options_lifetime);
-    RAY_CHECK(java_actor_lifetime != nullptr);
-    jint actor_lifetime_value =
-        env->GetIntField(java_actor_lifetime, java_actor_lifetime_value);
-    is_detached = (actor_lifetime_value == 1);
+    if (java_actor_lifetime != nullptr) {
+      jint actor_lifetime_value =
+          env->GetIntField(java_actor_lifetime, java_actor_lifetime_value);
+      RAY_CHECK_JAVA_EXCEPTION(env);
+      is_detached = std::make_shared<bool>(actor_lifetime_value == 1);
```

Comment:
Could you use something like `env->IsSameObject(java_actor_lifetime, STATUS_DETACHED)` instead of comparing the underlying values of enums?